apps/metagpt/cli.py

```diff
@@ -181,200 +118,134 @@ def metadata_new_str(self: Self) -> str:
     def metadata_old_str(self: Self) -> str:
         """Read a metadata file and returns its content."""
         if self.__metadata_old is None:
-            self.__metadata_old = _read_metadata_file(self.path_to_file)
+            self.__metadata_old = read_metadata_file(self.path_to_file)
         return self.__metadata_old
 
+    @property
+    def path_to_dataset(self: Self) -> str:
+        """Path to the dataset file based on the path to the metadata file."""
+        parts = self.path_to_file.split("/")
+        parts.remove("steps")
+        parts.remove("etl")
+        parts[-1] = parts[-1].split(".")[0]
+        return "/".join(parts)
+
+    def load_dataset(self: Self) -> Dataset:
+        """Load dataset.
+
+        Only works when channel is 'garden' or 'grapher'.
+        """
+        if self.channel not in [Channels.GARDEN, Channels.GRAPHER]:
+            raise Exception(f"Invalid channel. Should either be '{Channels.GARDEN}' or '{Channels.GRAPHER}'.")
+        # Check if the garden step file exists
+        if os.path.isfile(self.path_to_dataset + "/index.json"):
+            ds = Dataset(self.path_to_dataset)
+            # ds_meta_description = ds.metadata.to_dict()
+            return ds
+        else:
+            log.error(f"Required garden dataset {self.path_to_dataset} does not exist. Run the garden step first.")
+            raise Exception("Required garden dataset does not exist. Run the garden step first.")
+
+    def load_yaml_metadata(self: Self) -> Dict[str, Any]:
+        """Load dataset metadata from YAML."""
+        with open(self.path_to_file, "r") as file:
+            original_yaml_content = yaml.safe_load(file)
+        return original_yaml_content
+
     def save_updated_metadata(self: Self, output_file: str) -> None:
```

Comment:
in wizard this function never gets called but it works ok now from the terminal

Reply:
Saving the file differently on wizard. Probably the code in wizard is unnecessary complex, but that's the way I found to allow the user to further edit the GPT returned content.